demo/api/src/importer/pipes/importer-pipe.type.ts

```diff
@@ -3,6 +3,15 @@ import { ValidationError as ClassValidationError } from "class-validator";
 import { Transform } from "stream";
 
 export type PipeData = Record<string, unknown>;
+export type PipeMetadata<T> = {
+    temporalIdentifier?: string;
+    additionalData?: T;
```

Comment:
Now I see why we have the extra level 😁 Could we do `type PipeMetadata<T> = { temporalIdentifier?: string } & T` instead?